### ISSUE TITLE
Update dependency constraint to allow for rouge v4.0.0

### ISSUE THREAD

**Issue by jekyllbot** [jekyll/jekyll#9123]
Hey there! :wave:

I noticed that the constraint you have for rouge doesn't allow for the latest version to be used.

The constraint I found was `~> 3.0`, and the latest version available is `4.0.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

**Comment by 0xdf223** [comment]
Yes please. The HTTP2 syntax is fixed in v4 and would be really nice to have.

**Comment by ashmaroli** [comment]
@0xdf223 Did you try editing your Gemfile and loading that version directly?

**Comment by 0xdf223** [comment]
I did, but I get:
```
$ bundle install
Fetching gem metadata from https://rubygems.org/.........
Resolving dependencies...
Bundler could not find compatible versions for gem "rouge":
  In Gemfile:
    rouge (>= 4.0.0)

    jekyll was resolved to 4.2.2, which depends on
      rouge (~> 3.0)
```

**Comment by 0xdf223** [comment]
@ashmaroli - Is this something I should be able to get working without waiting on Jekyll?

**Comment by ashmaroli** [comment]
@0xdf223 Unfortunately, the workaround for this situation is a bit laborious. It involves creating and temporarily maintaining a Jekyll fork.
Firstly: 
- do you intend to develop your site on your local system and then push the changes onto GitHub or 
- develop your site directly on GitHub by pushing multiple commits?

If the first situation, then fork the Jekyll repository into your account and clone that fork into a location on your local system. Edit the `jekyll.gemspec` to allow Rouge 4.0:

```ruby
# jekyll.gemspec

s.add_runtime_dependency("rouge", "~> 4.0")
```

Then in your Jekyll project directory, edit the Gemfile to point to the cloned local Jekyll repository:
```ruby
# Gemfile

gem "jekyll", path: "<path_to_local_clone>"
```
Then delete existing `Gemfile.lock` and continue developing as usual:

```
bundle exec jekyll <build | serve>
```

The second situation with remote development is similar with changes, but you have to first commit the change to `jekyll.gemspec` and have it pushed to the fork on GitHub. Then edit your Gemfile on the Jekyll site repository to point to your GitHub fork:
```
gem "jekyll", github: "jekyll/jekyll", branch: "<name of branch with committed change>
```

**Comment by ashmaroli** [comment]
@0xdf223 We have patched the master branch to allow Rouge 4. If you'd like to use Rouge 4 now, you can do so by pointing your Gemfile to the commit:
```ruby
gem "jekyll", git: "https://github.com/jekyll/jekyll.git", ref: "1baded9"
```
(l*et me know in case the above doesn't work. It's been a while since I last used this syntax to pull a repo.*)

**Comment by 0xdf223** [comment]
Great news! Thank you.

Looks like I didn't have to update anything. My Gemfile was already showing:
```
# updated based on https://forestry.io/blog/how-i-reduced-my-jekyll-build-time-by-61/
gem "jekyll", github: "jekyll/jekyll"
```
I just did a `bundle update`, and rebuilt the site and it seem to be working locally. Going to push an update to Gitlab now.

**Comment by 0xdf223** [comment]
Looks great on my site! 
![image](https://user-images.githubusercontent.com/76954092/192799198-77417ca2-5671-4c05-b389-7a0fd42fdc80.png)

**Comment by ashmaroli** [comment]
Thank you for getting back to me @0xdf223
Happy Jekyllin' 😃
